tests/test_commands.py

```diff
@@ -339,10 +339,27 @@ class WindowsRunSpiderCommandTest(RunSpiderCommandTest):
     spider_filename = 'myspider.pyw'
 
     def setUp(self):
-        if platform != 'win32':
-            raise unittest.SkipTest(".pyw file extension only supported in Windows")
         super(WindowsRunSpiderCommandTest, self).setUp()
 
+    if platform != 'win32':
+        def test_run_good_spider(self):
+            raise unittest.SkipTest("Windows required")
+
+        def test_runspider(self):
+            raise unittest.SkipTest("Windows required")
+
+        def test_runspider_dnscache_disabled(self):
+            raise unittest.SkipTest("Windows required")
+
+        def test_runspider_log_level(self):
+            raise unittest.SkipTest("Windows required")
+
+        def test_runspider_log_short_names(self):
+            raise unittest.SkipTest("Windows required")
+
+        def test_runspider_no_spider_found(self):
+            raise unittest.SkipTest("Windows required")
+
```

Comment:
I think it would be better to use `@skipif` instead, in each method, and have their implementation be `super().<method_name>()`.

In any case, do all of those test fail on Linux?

Reply:
> I think it would be better to use `@skipif` instead, in each method, and have their implementation be `super().<method_name>()`.

For that I have to import `unittest` separately as `twisted.trial.unittest` doesn't have `@skipif`

Reply:
> In any case, do all of those test fail on Linux?

I don't think there is any case for it.

Reply:
If the tests are going to be Windows-only then, maybe this should wait until #4458 is merged. I’m hoping to be done with that soon™.

Reply:
That will be nice, lets wait then.

Reply:
Should I use `@skipIf` or current implementation is fine?

Reply:
I think you should use a decorator, `if` statements for method definition seem rather unidiomatic.

I wonder if you can use a skipping decorator or similar at the class level, though.